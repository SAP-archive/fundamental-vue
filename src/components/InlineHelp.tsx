import { Component, DefaultSlot, Prop, Base } from '@/core';

const positionMapping = {
  left: 'Left of icon',
  right: 'Right of icon (default)',
  center: 'Center of icon',
};

type Position = keyof (typeof positionMapping);
const Positions = Object.keys(positionMapping) as Position[];

interface Props {
  inline?: boolean;
  position?: Position;
}

@Component('InlineHelp')
@DefaultSlot('Helpful Text')
export class InlineHelp extends Base<Props> {
  @Prop('location of inline help relative to icon', { acceptableValues: Positions, type: String, default: 'right' })
  public position!: Position;

  @Prop('whether the inline help is displayed adjacent to the icon', { type: Boolean, default: false })
  public inline!: boolean;

  public render() {
    return (
    <span class='fd-inline-help'>
      <span class={this.classes}>
          {this.$slots.default}
      </span>
    </span>
    );
  }

  private get classes() {
    return {
      'fd-inline-help__content': true,
      'fd-inline-help__content--bottom-left': this.position === 'left' && this.inline === false,
      'fd-inline-help__content--bottom-right': this.position === 'right' && this.inline === false,
      'fd-inline-help__content--bottom-center': this.position === 'center' && this.inline === false,
      'fd-inline-help__content--left': this.position === 'left' && this.inline === true,
      'fd-inline-help__content--right': this.position === 'right' && this.inline === true,
    };
  }
}
