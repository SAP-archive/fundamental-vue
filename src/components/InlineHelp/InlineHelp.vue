<template>
  <span class="fd-inline-help">
    <span :class="classes"><slot /></span>
  </span>
</template>

<script lang="ts">
import Vue from "vue";
import { PropValidator } from "vue/types/options";

const positionMapping = {
  left: "left",
  right: "right", // default
  center: "center"
};

type Position = keyof (typeof positionMapping);
const Positions = Object.keys(positionMapping);
const isValidPosition = (value: string) => Positions.indexOf(value) >= 0;

export default Vue.extend({
  name: "FdInlineHelp",
  props: {
    position: {
      type: String,
      default: positionMapping.right,
      validator: isValidPosition
    } as PropValidator<Position>,
    inline: { type: Boolean, default: false }
  },
  computed: {
    classes(): object {
      return {
        "fd-inline-help__content": true,
        "fd-inline-help__content--bottom-left":
          this.position === "left" && this.inline === false,
        "fd-inline-help__content--bottom-right":
          this.position === "right" && this.inline === false,
        "fd-inline-help__content--bottom-center":
          this.position === "center" && this.inline === false,
        "fd-inline-help__content--left":
          this.position === "left" && this.inline === true,
        "fd-inline-help__content--right":
          this.position === "right" && this.inline === true
      };
    }
  }
});
</script>
