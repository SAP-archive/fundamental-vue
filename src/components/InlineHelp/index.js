import InlineHelp from "./InlineHelp.vue";
import { pluginify } from "./../../util";
export default pluginify(InlineHelp);
export { InlineHelp };
