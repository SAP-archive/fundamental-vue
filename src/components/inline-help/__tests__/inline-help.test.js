import { mount } from '@vue/test-utils'
import InlineHelp from '../inline-help.vue'

describe('InlineHelp test scripts', () => {
  test('Renders correctly', () => {
    const inlineHelp = mount(InlineHelp, {})
    expect(inlineHelp.element).toMatchSnapshot()
  })

  test('Default slot test', () => {
    const helpText = 'Slot Text'
    const inlineHelp = mount(InlineHelp, {
      slots: {
        default: helpText
      }
    })
    expect(inlineHelp.element).toMatchSnapshot()
    expect(inlineHelp.text()).toBe(helpText)
  })
})
