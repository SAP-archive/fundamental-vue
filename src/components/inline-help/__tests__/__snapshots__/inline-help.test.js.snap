// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineHelp test scripts Default slot test 1`] = `
<span
  class="fd-inline-help"
>
  <span
    class="fd-inline-help__content fd-inline-help__content--bottom-right"
  >
    Slot Text
  </span>
</span>
`;

exports[`InlineHelp test scripts Renders correctly 1`] = `
<span
  class="fd-inline-help"
>
  <span
    class="fd-inline-help__content fd-inline-help__content--bottom-right"
  />
</span>
`;
